### PR TITLE
Fix default value for dictionary byte column

### PR DESCRIPTION
**Pull request by jackjlli**
This PR fixes default value for dictionary byte column.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java [hunk 1]
```diff
@@ -208,7 +208,7 @@ public long getLatestIngestionTimestamp() {
         // each forward index entry will contain a 4 byte dictionary ID
         forwardIndexColumnSize = FieldSpec.DataType.INT.size();
         int dictionaryColumnSize;
-        if (dataType == FieldSpec.DataType.STRING) {
+        if (!isFixedWidthColumn) {
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java]: better switch the if/else.

also, good to add BYTES column in the schema for realtime test (can be another PR)

**Comment by jackjlli** [comment]
> Please change RawIndexConverter line 168 for the same issue

Good catch, updated.

**Comment by siddharthteotia** [comment]
@jlli, thanks for fixing this. I think the PR description is misleading. It is a dictionary column. Also I have an open PR for adding tests. https://github.com/apache/incubator-pinot/pull/5083 We can add more tests in that PR
